org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/generator/trace/TraceForStorageProvider.java

```diff
@@ -69,7 +69,7 @@ public AbsoluteURI getPath() {
 
 		@Override
 		public long getTimestamp() {
-			return file.getLocalTimeStamp();
+			return file.getModificationStamp();
```

Comment:
Hmm I'm not sure.

Reply:
@szarnekow I haven't tried, but the Javadoc of `getLocalTimeStamp`, https://help.eclipse.org/latest/index.jsp?topic=%2Forg.eclipse.platform.doc.isv%2Freference%2Fapi%2Forg%2Feclipse%2Fcore%2Fresources%2FIResource.html&anchor=getLocalTimeStamp(), also says

> Note that due to varying file system timing granularities, this value is not guaranteed to change every time the file is modified. For a more reliable indication of whether the file has changed, use [getModificationStamp()](https://help.eclipse.org/latest/topic/org.eclipse.platform.doc.isv/reference/api/org/eclipse/core/resources/IResource.html#getModificationStamp()).

It sounds like the method is quite unreliable...

Reply:
i guess a sleep of 1s would also help. but am not sure if other tests might be affected too

Reply:
> Hmm I'm not sure.

@szarnekow what do you think is wrong with `getModificationStamp`?

Reply:
It's not a timestamp and thus *should* not be used to implement `getTimestamp`

It might be fine to rename getTimestamp to getModification stamp

Reply:
> It's not a timestamp and thus _should_ not be used to implement `getTimestamp`
> 
> It might be fine to rename getTimestamp to getModification stamp

@szarnekow OK, I'll do that later.

That would be API breakage but I guess for that wouldn't be a problem because it's an internal type, would it?

Reply:
@szarnekow renamed it (https://github.com/eclipse-xtext/xtext/pull/3354/commits/7cc6b0d715b354580438c2f0fc0cafd07b627b80)